### PR TITLE
Add options to opt into advanced pom support and into gradle metadata

### PR DESCRIPTION
**Pull request by jjohannes**
See: #3945
 
Note: the first two commits turn the existing _experimental_ flag into an option. Which is then superseded by the two new options in the other commits.

Note: This **depends on #3870** to be merged first (that should resolve the conflicts this PR currently has with master).

### CODE REVIEW

File: subprojects/core-api/src/main/java/org/gradle/StartParameter.java [hunk 1]
```diff
@@ -850,4 +851,24 @@ public boolean isInteractive() {
     public void setInteractive(boolean interactive) {
         this.interactive = interactive;
     }
+
+    /**
+     * Returns true when optional dependencies, compile/runtime scope separation and BOMs are included in dependency resolution.
+     *
+     * @since 4.5
```
Comment (jjohannes) [inline subprojects/core-api/src/main/java/org/gradle/StartParameter.java]: Note: This will be changed into 4.6 once the 4.5 branching is completed.

**Comment by melix** [comment]
Do we really want to introduce things like `--advanced-pom-support` as CLI options? This looks pretty wrong to me as it's something transient. I'm a bit confused.

File: subprojects/dependency-management/preview-features.adoc [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
+# Gradle 5.0 Feature Previews
 
 ## Enhanced support for Maven repositories
 
-Because these improvements change the resolution result for certain Gradle `Configuration` instances, we consider them to be "potentially breaking changes". As such, these improvements are _opt-in_ for the Gradle 4.x stream of releases, but we expect to make them the default in Gradle 5.0.
+These improvements can change the resolution result for certain Gradle configurations. Thus we consider them to be _potentially breaking changes_. As such, these improvements are opt-in for the Gradle 4.x stream of releases, but are scheduled to become the default in Gradle 5.0.
 
-### Dependency constraints from Maven POM files
+[NOTE]
+Activate this feature in Gradle 4.6+ by setting `org.gralde.advancedpomsupport=true` in _gradle.properties_ or calling Gradle with
```
Comment (ldaley) [inline subprojects/dependency-management/preview-features.adoc]: Typo in sysprop name here.

**Comment by ldaley** [comment]
Invocation time switches for this doesn't make a lot of sense to me (in addition to the other reasons not to do this). Whether to use these features or not is an attribute of the build, not how it is invoked.

**Comment by jjohannes** [comment]
Thanks for having a look at this @ldaley. I also had more discussion about this with @melix and @bigdaz. And we agree.

I will remove the switches `--something` so that we only have the properties.

The initial idea of having the ability to do this from command line was to allow users to easily try out a preview feature to test if it changes/breaks their build. But eventually it should always become part of the build configuration. After removing the options, you will still be able to set the property from the command line, which should be enough for the "try it out" use case (e.g. `-Dorg.gradle.advancedpomsupport=true`).

**Pull request merged**